src/lib/input/input-container.ts

```diff
@@ -164,7 +157,10 @@ export class MdInputDirective implements AfterContentInit {
 
   _onBlur() { this.focused = false; }
 
-  _onInput() { this.value = this._elementRef.nativeElement.value; }
+  _onInput() {
+    // This is a noop function and is used to let Angular recognize the changes
+    // to the input element while typing. Listening to the `input` event is similar to NgControls.
```

Comment:
Nitpick: Can you add some more detail here?  I'd also add that if you're using one of the FormsModule or ReactiveFormsModule directives, the input event is already watched and will kick off change detection appropriately. But that if you're not using one of these and you've disabled the floating placeholder, we need to ensure that change detection still runs immediately on input to hide the placeholder.